src/test/java/kitchenpos/application/ProductServiceTest.java

```diff
@@ -0,0 +1,177 @@
+
+    private ProductService productService;
+
+    @BeforeEach
+    void setUp() {
+        productService = new ProductService(productRepository, menuRepository, purgomalumClient);
+    }
+
+    @DisplayName("모든 `상품`를 조회할 수 있다.")
+    @Test
+    void findAllProducts() {
+        productRepository.save(
+            ProductFixture.createProduct("HOT 후라이드 치킨", BigDecimal.valueOf(21_000L)));
+        productRepository.save(ProductFixture.createProduct("양념 치킨", BigDecimal.valueOf(22_000L)));
+
+        assertThat(productService.findAll()).hasSize(2);
+    }
+
+    @Nested
+    class CreateProductTests {
+
+        @DisplayName("`상품`을 생성할 수 있다.")
+        @Test
+        void createProductWithValidInput() {
+            Product validProduct = ProductFixture.createProduct();
+
+            var product = productService.create(validProduct);
+
+            assertAll(
+                () -> assertThat(product).isNotNull(),
+                () -> assertThat(product.getId()).isNotNull(),
+                () -> assertThat(product.getName()).isEqualTo(validProduct.getName()),
+                () -> assertThat(product.getPrice()).isEqualTo(validProduct.getPrice())
+            );
+        }
+
+        @DisplayName("`상품`의 가격은 0 이상이어야 한다")
+        @Test
+        void createProductWithNegativePrice() {
+            Product invalidRequest = ProductFixture.createProduct();
+            invalidRequest.setPrice(BigDecimal.valueOf(-1000));
+
+            assertThrows(IllegalArgumentException.class,
+                () -> productService.create(invalidRequest));
+        }
+
+        @DisplayName("`상품`의 가격은 null이 아니어야 한다")
+        @Test
+        void createProductWithNullPrice() {
+            Product request = ProductFixture.createProduct();
+            request.setPrice(null);
+
+            assertThrows(IllegalArgumentException.class, () -> productService.create(request));
+        }
+
+        @DisplayName("`상품`의 이름은 null이 아니어야 한다")
+        @Test
+        void createProductWithNullName() {
+            Product request = ProductFixture.createProduct();
+            request.setName(null);
+
+            assertThrows(IllegalArgumentException.class, () -> productService.create(request));
+        }
+
+        @DisplayName("`상품`의 이름에 욕설이 포함되어 있으면 생성할 수 없다")
+        @Test
+        void createProductWithProfanityInName() {
+            Product request = ProductFixture.createProduct();
+            request.setName("대충 나쁜 말");
+
+            assertThrows(IllegalArgumentException.class, () -> productService.create(request));
+        }
+    }
+
+    @Nested
+    class ChangeProductTests {
+
+        @DisplayName("`상품`의 가격을 변경할 수 있다")
+        @Test
+        void changeProductPriceWithValidInput() {
+            Product existingProduct = ProductFixture.createProduct();
+
+            MenuProduct menuProduct = MenuProductFixture.createMenuProduct(existingProduct);
+
+            Menu menu = MenuFixture.createMenu();
+            menu.setMenuProducts(List.of(menuProduct));
```

Comment:
테스트는 잘 작성해주셔서 대다수는 비슷한 코멘트를 남기게 될 것 같네요. 🙂 (사실 이 부분만 수정해주시면 될 것 같은데, 


아래와 같은 느낌처럼 fixture 함수를 호출한 결과 그대로 테스트에서는 사용하는 것이 조금 더 테스트가 깔끔해질 것 같아요. 
지금은 테스트 케이스에서 fixture를 생성하고 또 fixture를 의도한 값으로 set 해주는 로직들이 많다보니 테스트의 핵심적인 부분이 한눈에 잘 안보이는 것 같아요. 🤔

```suggestion
            Menu menu = MenuFixture.createMenu(List.of(menuProduct));
```